packages/block-library/src/image/view-interactivity.js

```diff
@@ -17,293 +17,426 @@ const focusableSelectors = [
 	'[tabindex]:not([tabindex^="-"])',
 ];
 
-store( {
-	actions: {
-		core: {
-			image: {
-				showLightbox: ( { context, event } ) => {
-					// We can't initialize the lightbox until the reference
-					// image is loaded, otherwise the UX is broken.
-					if ( ! context.core.image.imageLoaded ) {
-						return;
-					}
-					context.core.image.initialized = true;
-					context.core.image.lastFocusedElement =
-						window.document.activeElement;
-					context.core.image.scrollDelta = 0;
-
-					context.core.image.lightboxEnabled = true;
-					if ( context.core.image.lightboxAnimation === 'zoom' ) {
-						setZoomStyles(
-							event.target.nextElementSibling,
-							context,
-							event
-						);
-					}
-					// Hide overflow only when the animation is in progress,
-					// otherwise the removal of the scrollbars will draw attention
-					// to itself and look like an error
-					document.documentElement.classList.add(
-						'has-lightbox-open'
-					);
-
-					// Since the img is hidden and its src not loaded until
-					// the lightbox is opened, let's create an img element on the fly
-					// so we can get the dimensions we need to calculate the styles
-					context.core.image.preloadInitialized = true;
-					const imgDom = document.createElement( 'img' );
-					imgDom.onload = function () {
-						context.core.image.activateLargeImage = true;
-					};
-					imgDom.setAttribute(
-						'src',
-						context.core.image.imageUploadedSrc
-					);
-				},
-				hideLightbox: async ( { context, event } ) => {
-					context.core.image.hideAnimationEnabled = true;
-					if ( context.core.image.lightboxEnabled ) {
-						// If scrolling, wait a moment before closing the lightbox.
-						if ( context.core.image.lightboxAnimation === 'fade' ) {
-							context.core.image.scrollDelta += event.deltaY;
-							if (
-								event.type === 'mousewheel' &&
-								Math.abs(
-									window.scrollY -
-										context.core.image.scrollDelta
-								) < 10
-							) {
-								return;
-							}
-						} else if (
-							context.core.image.lightboxAnimation === 'zoom'
-						) {
-							// Disable scroll until the zoom animation ends.
-							// Get the current page scroll position
-							const scrollTop =
-								window.pageYOffset ||
-								document.documentElement.scrollTop;
-							const scrollLeft =
-								window.pageXOffset ||
-								document.documentElement.scrollLeft;
-							// if any scroll is attempted, set this to the previous value.
-							window.onscroll = function () {
-								window.scrollTo( scrollLeft, scrollTop );
-							};
-							// Enable scrolling after the animation finishes
-							setTimeout( function () {
-								window.onscroll = function () {};
-							}, 400 );
+store(
+	{
+		state: {
+			windowWidth: window.innerWidth,
+			windowHeight: window.innerHeight,
```

Comment:
I don't know if we should use namespaces here as well. It's true that it could be used by any other block, but it could also create conflicts if other blocks declare the same variables.

Reply:
Good point — I [pushed a commit](https://github.com/WordPress/gutenberg/pull/52765/commits/b9c84e48b0df2fd0e45588f7eda960132aeed900) adding a namespace.